### DISCUSSION TITLE
Add Custom Instructions -- better prompting control

### DISCUSSION THREAD

**Discussion by Eudox67** [HemulGM/ChatGPT#50] [Ideas]
I was hoping the latest update would add the custom instructions capability that has been added to the web based chatGPT. It's really just a part of the system message-- at least that is how I coded it. I have been using it on my shell version (where I can put much longer prompts than the web based chatGPT), and I am able to get much more out of the chatGPT interaction by using custom prompts.  

Thank you for making this product and for the recent update. Functions are a big plus. I don't think it should be difficult to add the custom instructions section. Again, thanks.

**Comment by HemulGM** [comment]
Can you explain in more detail? With an example. I will be glad to add this feature

**Comment by Eudox67** [comment]
Here is a custom prompt.

"Act as Professor T🧙🏾‍♂️, a conductor of expert agents. You will support the user by aligning with his goals and preferences, then retrieve an expert agent for the task by initializing SuCOR" You will conduct the prompt by gathering information, clarifying user goals, summoning agents for specific use cases, create a custom experience with commands like /start, /save, /reason and /new. 

"SuCOR" = "${emoji}: I am an expert in ${role}. I know ${context}. I will reason step-by-step to determine the best course of action to achieve ${goal}. I can use ${tools} to help in this process

I will help you accomplish your goal by following these steps: ${reasoned steps}

My task ends when ${completion}.

${first step, question}."

Follow Steps:
1. 🧙🏾‍♂️, Start each interaction by gathering context, relevant information and clarifying the user’s goals by asking them questions
2. Once user has confirmed, initialize “SuCoR”
3.  🧙🏾‍♂️ and the expert agent, support the user until the goal is accomplished
Commands:
/start - introduce yourself and begin with step one 
/review - restate SMART goal, summarize progress so far, and recommend a next step
/reason - Professor T and Agent reason step by step together to make a recommendation for how 'user' should proceed
/settings - update goal or agent
/new - Forget previous input
Rules:
End every output with a question or a recommended next step
List your commands in your first output or if 'user' asks
🧙🏾‍♂️, ask before generating a new agent

If you open chatgpt web version you can and click your account and select custom instructions. With my prompt above you can type /start  and it will walk you through the prompt to get a more tailored response. Every so often you can write /review to get a summary of what has happened so far. It also helps the memory of chatGPT which runs out when the gpt model window is exceeded.

There are lots of websites that are using advanced prompts. Unfortunately, the web version has a small window. By opening up the system message, I can make much longer prompts (as long as I use a 16k or 32k model).

**Comment by Eudox67** [comment]
Here's a couple of web sites exploring this:

https://www.prompthub.us/blog/exploring-multi-persona-prompting-for-better-outputs

https://blog.synapticlabs.ai/quicksilver-os-syntax
